src/control/server/engine/utils.go

```diff
@@ -51,17 +52,98 @@ const (
 	logMasksStrAssignOp  = "="
 )
 
-var validLogLevels = []string{
-	"DEBUG", "DBUG", "INFO", "NOTE", "WARN", "ERROR", "ERR", "CRIT", "ALRT", "EMRG", "EMIT",
+// LogLevel enum representing engine logging levels.
+type LogLevel uint
+
+// LogLevels matching D_LOG API priority strings.
+const (
+	LogLevelUndefined LogLevel = iota
+	LogLevelDbug
+	LogLevelInfo
+	LogLevelNote
+	LogLevelWarn
+	LogLevelErr
+	LogLevelCrit
+	LogLevelAlrt
+	LogLevelEmrg
+	LogLevelEmit
+)
+
+func (ll LogLevel) String() string {
+	switch ll {
+	case LogLevelDbug:
+		return "DBUG"
+	case LogLevelInfo:
+		return "INFO"
+	case LogLevelNote:
+		return "NOTE"
+	case LogLevelWarn:
+		return "WARN"
+	case LogLevelErr:
+		return "ERR"
+	case LogLevelCrit:
+		return "CRIT"
+	case LogLevelAlrt:
+		return "ALRT"
+	case LogLevelEmrg:
+		return "EMRG"
+	case LogLevelEmit:
+		return "EMIT"
+	default:
+		return ""
+	}
 }
 
+func StrToLogLevel(s string) LogLevel {
```

Comment:
nit - needs documentation comment

Reply:
will address on the next push or in a follow-up PR

Reply:
done